### PR TITLE
implement tag filtration

### LINKED ISSUE #846: Tags navigation not filtering products anymore
It worked fine on 0.10 and 0.11.
Since 0.12 and with latest changes from development branch, filtering by tags doesn't happen anymore.
All products are always displayed.

### LINKED ISSUE #870: Receive server exception on startup in translations pub
From a fresh clone, after doing a `reaction reset` I get the error below because `null` is being passed in.  Afterwards it chatters back and forth between `en-us` and `en`.

Steps taken to try and isolate an environmental issues:
1. Switched to `v0.11.1` branch, error disappears.
2. Using VPN, switched Locale to US, issue still occurs although the toggling between `en` and `en-us` stopped. I always got `en-US`.
3. Only seems to happen maybe once or twice after reset.
 
```
Exception from sub Translations id bfr8puxEcDkmJo7r4 Error: Match error: Expected string, got object
    at check (packages/check/match.js:33:1)
    at [object Object].ReactionSubscriptions (server/publications/translations.js:7:3)
    at packages/matb33_collection-hooks/collection-hooks.js:275:1
    at [object Object]._.extend.withValue (packages/meteor/dynamics_nodejs.js:56:1)
    at [object Object]._handler (packages/matb33_collection-hooks/collection-hooks.js:274:1)
    at packages/check/match.js:103:1
    at [object Object]._.extend.withValue (packages/meteor/dynamics_nodejs.js:56:1)
    at Object.Match._failIfArgumentsAreNotAllChecked (packages/check/match.js:102:1)
    at maybeAuditArgumentChecks (livedata_server.js:1695:18)
    at [object Object]._.extend._runHandler (livedata_server.js:1023:17)
    at [object Object]._.extend._startSubscription (livedata_server.js:842:9)
    at [object Object]._.extend.protocol_handlers.sub (livedata_server.js:614:12)
    at livedata_server.js:548:43
Sanitized and reported to the client as: Match failed [400]
```

### PR DESCRIPTION
**Pull request by aaronjudd**
Resolves #846
Resolves #870

- ReactionProduct.getProductPriceRange().priceRange updated to return
range, min, max.
- denormalize function updated to add price range, min, max
- PriceRange added to product schema
- Add filtering to the tags, as originally conceived for
https://github.com/reactioncommerce/reaction/pull/743
— allow for multiple tags in query params

The format is:
```
http://localhost:3000/reaction/tag/shop?price.min=100&price.max=200
```
regex text search of title, description:
```
http://localhost:3000/?query=%22Copy%2010%22
```

### CODE REVIEW

**Comment by aaronjudd** [comment]
This is now complete and ready for merge. All tests passing.

File: packages/reaction-catalog/server/methods.js [hunk 1]
```diff
@@ -196,12 +196,11 @@ function denormalize(id, field) {
       isLowQuantity: isLowQuantity(variants)
     });
     break;
-  default: // "price"
+  default: // "price" is object with range, min, max
     // set "0" if no variants in product. If all variants were removed.
-    const priceRange = ReactionCore.getProductPriceRange(id) || 0;
-    Object.assign(update, { price: priceRange });
+    const priceObject = ReactionCore.getProductPriceRange(id) || 0;
```
Comment (zenweasel) [inline packages/reaction-catalog/server/methods.js]: this won't cause an issue if this isn't an object but an integer? (if it's 0)

Comment (aaronjudd) [inline packages/reaction-catalog/server/methods.js]: yeah, agreed the defaults should be removed.

**Comment by ramusus** [comment]
@aaronjudd I see in this PR has many ideas and some code from this  PR https://github.com/reactioncommerce/reaction/pull/743 
Do you intend to implement `filtration` as external package? If yes I'm ready to updated this package for last development and make new PR for supporting flattened schema. I was thinking to make dependancy from `tmeasday:publish-counts` optional or just for admins.

And question about price range attribute - do we still need it in DB in case that we can always generate it on the fly from price min and max values.

**Comment by newsiberian** [comment]
The goal of using denormalized price range inside doc is to get speed benefit from it.  `getProductPriceRange` is a function with a number of computations.
I suggest to think about product grid how about bottleneck of all reaction. We should do all we can to make in work smooth and fast. This is what I believe.

**Pull request merged**